src/sentry/search/events/datasets/spans_indexed.py

```diff
@@ -822,32 +839,46 @@ def _resolve_unadjusted_margin(
 
     def _resolve_finite_population_correction(
         self,
+        args: Mapping[str, str | Column | SelectType | int | float],
         total_samples: SelectType,
         population_size: int | float,
     ) -> SelectType:
-        return Function(
-            "sqrt",
-            [
-                Function(
-                    "divide",
-                    [
-                        Function("minus", [population_size, total_samples]),
-                        Function("minus", [population_size, 1]),
-                    ],
-                )
-            ],
+        return (
+            Function(
+                "sqrt",
+                [
+                    Function(
+                        "divide",
+                        [
+                            Function("minus", [population_size, total_samples]),
+                            Function("minus", [population_size, 1]),
+                        ],
+                    )
+                ],
+            )
+            if args["fpc"] == "fpc"
+            else 1
         )
 
     def _resolve_lower_limit(
         self,
         args: Mapping[str, str | Column | SelectType | int | float],
         alias: str,
     ) -> SelectType:
-        total_samples, _ = self._query_total_counts()
+        _, total_proportion = self._query_total_counts()
         sampled_group = Function("count", [])
-        proportion_by_sample = Function("divide", [sampled_group, total_samples])
-        return Function(
+        proportion_by_sample = Function(
             "divide",
+            [
+                sampled_group,
+                Function(
+                    "multiply", [total_proportion, Function("avg", [Column("sampling_factor")])]
+                ),
+            ],
+            "proportion_by_sample",
+        )
+        return Function(
+            "round",
```

Comment:
These functions are a bit hard to review due to the Snuba syntax - could you add some docstrings or similar with regular mathematical notation to indicate what certain functions are / should be calculating?